File: app/src/main/java/com/example/pam_project/features/tasks/list/TaskView.java
package com.example.pam_project.features.tasks.list;

import java.util.List;

public interface TaskView {

    void bindTasks(final List<TaskInformation> model);

    void bindHeaders(int[] headers);

    void bindListName(final String name);

    void showAddTask();

    void onTaskStatusEdit(final TaskInformation model, final int adapterPosition);

    void showTaskContent(final long id);

    void showEmptyMessage();

    void showEditList();
}

File: app/src/main/java/com/example/pam_project/features/tasks/list/TaskView.kt
package com.example.pam_project.features.tasks.list

interface TaskView {
    fun bindTasks(model: List<TaskInformation?>?)
    fun bindHeaders(headers: IntArray)
    fun bindListName(name: String?)
    fun showAddTask()
    fun onTaskStatusEdit(model: TaskInformation?, adapterPosition: Int)
    fun showTaskContent(id: Long)
    fun showEmptyMessage()
    fun showEditList()
    fun onTasksReceivedError()
    fun onTaskUpdatedError()
}